src/editor/CodeHintManager.js

```diff
@@ -410,7 +412,48 @@ define(function (require, exports, module) {
         }
         return false;
     }
+    function _callMoveUp(event) {
+        if (deferredHints) {
+            deferredHints.reject();
+            deferredHints = null;
+        }
 
+        var response = sessionProvider.getHints(lastChar);
+        lastChar = null;
+
+        if (!response) {
+            // the provider wishes to close the session
+            _endSession();
+        } else {
+            // if the response is true, end the session and begin another
+            if (response === true) {
+                var previousEditor = sessionEditor;
+                //_endSession();
+                //_beginSession(previousEditor);
+            } else if (response.hasOwnProperty("hints")) { // a synchronous response
+                if (hintList.isOpen()) {
+                    // the session is open
+                    hintList.callMoveUp(event);
+                }
+            } else { // response is a deferred
```

Comment:
@abose @petetnt this is actually just the function _updateHintList() copied over and some if statements removed. Do you think it would be better to just check inside _updateHintList if codeHintOpened and then call _moveUpList inside there?